lib/util/calculation.dart

```diff
@@ -0,0 +1,281 @@
+  exponent: const ExponentCalculation(),
+  exponent_symbol: const ExponentCalculation(),
+
+  factorial: const FactorialCalculation(),
+  factorial_symbol: const FactorialCalculation(),
+
+  logOfX: const LogCalculation(),
+
+  multiply: const MultiplyCalculation(),
+
+  naturalLog: const NaturalLogCalculation(),
+  nthRootSymbol: const NthRootCalculation(),
+
+  percent_symbol: const PercentCalculation(),
+
+  sine: const SineCalculation(),
+  subtract: const SubtractCalculation(),
+  sqrtSymbol: const SqrtCalculation(),
+  squared: const SquaredCalculation(),
+
+  tangent: const TangentCalculation(),
+  tenToPower: const BaseTenCalculation(),
+
+  zerosShort: const ZerosCalculation()
+};
+
+getCalculation(String key) => _calculations[key] ?? key;
+
+
+class AddCalculation extends BinaryCalculation
+    implements BasicCalculation {
+
+  const AddCalculation() : super(Precedence.i);
+
+  num call(num x, num y) => x + y;
+}
+
+class ArcCosineCalculation extends UnaryCalculation {
+
+  const ArcCosineCalculation() : super(Precedence.v);
+
+  num call(num x) => Calculation.arcTrigWrapper(acos)(x);
+}
+
+class ArcSineCalculation extends UnaryCalculation {
+
+  const ArcSineCalculation() : super(Precedence.v);
+
+  num call(num x) => Calculation.arcTrigWrapper(asin)(x);
+}
+
+class ArcTangentCalculation extends UnaryCalculation {
+
+  const ArcTangentCalculation() : super(Precedence.v);
+
+  num call(num x) => Calculation.arcTrigWrapper(atan)(x);
+}
+
+
+class BaseTenCalculation extends UnaryCalculation {
+
+  const BaseTenCalculation() : super(Precedence.v);
+
+  num call(num exponent) => pow(10, exponent);
+}
+
+class BaseECalculation extends UnaryCalculation {
+
+  const BaseECalculation() : super(Precedence.v);
+
+  num call(num exponent) => pow(E, exponent);
+}
+
+
+class CosineCalculation extends UnaryCalculation {
+
+  const CosineCalculation() : super(Precedence.v);
+
+  num call(num x) => Calculation.trigWrapper(cos)(x);
+}
+
+
+class DivideCalculation extends BinaryCalculation
+    implements BasicCalculation {
+
+  const DivideCalculation() : super(Precedence.ii);
+
+  num call(num x, num y) => x / y;
+}
+
+
+class ExponentCalculation extends BinaryCalculation
+    implements BasicCalculation {
+
+  const ExponentCalculation() : super(Precedence.iv);
+
+  num call(num x, num y) => pow(x, y);
+}
+
+
+class FactorialCalculation extends UnaryCalculation {
+
+  const FactorialCalculation() : super(Precedence.iv);
+
+  num gamma(num x) {
+    return sqrt(2 * PI / x) *
+        pow((1 / E) * (x + 1 / (12 * x - 1 / (10 * x))), x);
+  }
+
+  num call(num x) {
+    if (x.toString().contains(".")) {
+      return gamma(x + 1);
+    }
+
+    int total = 1;
+    while (x > 1) {
+      total *= x--;
+    }
+    return total;
+  }
+}
+
+
+class LogCalculation extends UnaryCalculation {
+
+  const LogCalculation() : super(Precedence.v);
+
+  num call(num x) => log(x) / log(10);
+}
+
+
+class MultiplyCalculation extends BinaryCalculation
```

Comment:
Minor: Perhaps MultiplicationCalculation? 

Subtract -> Subtraction
Add -> Addition
Divide -> Division

Just flows better, in my opinion.